### PR TITLE
Accommodate rtcm mixed msg stream

### LINKED ISSUE #54: Skipping RTCM messages
In the log files I work with there are RTCM3 messages interspersed with UBX and NMEA. This results in broken iteration when those messages are encountered.

I've made this change to skip these RTCM3 messages (and return their raw data)

[rtcm3_skip_p2.patch.txt](https://github.com/semuconsulting/pyubx2/files/8034927/rtcm3_skip_p2.patch.txt)

Message format hint taken from here: https://github.com/tomojitakasu/RTKLIB/blob/71db0ffa0d9735697c6adfd06fdf766d0e5ce807/src/rtcm.c#L254

### PR DESCRIPTION
**Pull request by semuadmin**
# pyubx2 Pull Request Template

## Description

Add functionality in UBXReader.read() method to accommodate any RTCM3 data in a mixed protocol input stream. Any RTCM3 protocol messages in the input stream will be read _but not parsed_.

Add RTCM3 protocol filter value to keyword arguments.

Fixes #54 

## Testing

Please test all changes, however trivial, against the supplied unittest suite `tests/test_*.py` e.g. by executing the `tests/testsuite.py` module or using your IDE's native Python unittest integration facilities.
Please describe any test cases you have amended or added to this suite to maintain >= 99% code coverage.

- [x] Two tests added to test_stream.py
- [x] One test updated in test_static.py

## Checklist:

- [x] My code follows the style guidelines of this project (see `CONTRIBUTING.MD`).
- [x] I have performed a self-review of my own code.
- [x] (*if appropriate*) I have cited my u-blox documentation source(s).
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] (*if appropriate*) I have added test cases to the `tests/test_*.py` unittest suite to maintain >= 99% code coverage.
- [x] I have tested my code against the full `tests/test_*.py` unittest suite.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I understand and acknowledge that the code will be published under a BSD 3-Clause license.

### CODE REVIEW

**Review by semudev2 — COMMENTED** [review]
Just a suggestion, but might be worth adding provision in `gnssdump.py` for an external RTCM3 handler - similar to the existing provision for external NMEA or UBX handlers, for users who might have a Python RTCM3 parser to hand.

File: pyubx2/ubxreader.py [hunk 1]
```diff
@@ -200,6 +205,47 @@ def _parse_nmea(self, hdr: bytes) -> tuple:
...
+        :return: tuple of (raw_data as bytes, parsed_stub as string)
+        :rtype: tuple
+        """
+
+        hdr3 = self._read_bytes(1)
+        size = hdr3[0] | (hdr[1] << 8)
+        payload = self._read_bytes(size)
+        crc = self._read_bytes(3)
+        raw_data = hdr + hdr3 + payload + crc
+        parsed_stub = f"<RTCM3(data={payload}, crc={crc})>"
```
Comment (flytrex-vadim) [inline pyubx2/ubxreader.py:232]: not sure embedding the payload here is very useful - it is long, and readily accessible from the raw_data variable. What might be useful here is the RTCM message type, extractable as (payload[0] << 4 | payload[1] >> 4)

Comment (semuadmin) [reply pyubx2/ubxreader.py]: Agreed. I only had to hand an old (2006) copy of the RTCM 10403.1 standard and wasn't sure if the message type was _always_ the first 12 bits of the payload, but assuming it is then fine.

As an aside,  this old copy of the standard also states...

"The Preamble is a fixed 8-bit sequence. The next six bits are reserved in RTCM10403.1 and should be set to zero by the Transport Layer Control for all messages. _The mobile user receiver should ignore these bits and not assume they will always be set to zero_. In future versions these bits may contain the version number of the standard."

... which was partly why I'd avoided attempting to parse RTCM messages in the past, but it works for the time being at least.

I've added a stub `RTCMMessage` class definition as this facilitates the consistent use of message identity and protocol filters, etc.

**Review by semudev2 — APPROVED** [review]

**Comment by flytrex-vadim** [comment]
I'll put it here in case you'll ever want to implement CRC check on the RTCM messages - a working python implementation of CRC24Q :)

```
        def crc24q(octets):
            INIT = 0
            POLY = 0x1864CFB
            crc = INIT
            for octet in octets:
                crc ^= (octet << 16)
                for i in range(8):
                    crc <<= 1
                    if crc & 0x1000000: crc ^= POLY
            return crc & 0xFFFFFF
```

**Comment by semuadmin** [comment]
Duly noted 👍. Thanks for the donation by the way - appreciated.

**Pull request merged**